Clarify scoping for component names (#2591)
modelica/ModelicaSpecification: chapters/scoping.tex

\autoref{redeclaration} for further explanations regarding looking inside
partial packages.
\end{nonnormative}
\begin{example}

> Components and classes are part of the same name-space and thus a component may
> not have the same name as its class or the first part of the class-name as that

Reviewer: Just to clarify a tiny bit that this is an example showing a consequence, as opposed to an example containing a rule:
```suggestion
Components and classes are part of the same name-space and thus a component cannot
have the same name as its class or the first part of the class-name as that
```

> would prevent looking up the name of the class.

Reviewer: I'd rather rephrase this using the term _lookup_; rather than counting the two existing occurrences of _looking up_ as proof that we also use this term for the same thing, I think that those two occurrences should also be rewritten.  Perhaps like this:
```suggestion
would prevent lookup of the class name.
```

\begin{lstlisting}[language=modelica]
model A
  M M;    // Illegal, component 'M' prevents finding class 'M'
  P.Q P;  // Illegal, component 'P' prevents finding package 'P'
  .R R;   // Legal, see next section
  S.Q Q;  // Legal
end A;
\end{lstlisting}

> \end{example}

Reviewer 2: I like the example. It really clarifies. I wonder if the text and the example can be extended to also include cases of model C and model D in #2547.

Components and classes are part of the same name-space and thus a component may not have the same name as its class or the class of any other component in the same scope or the first part of the class-name....
